### PR TITLE
Test is_valid_hdr

### PR DESCRIPTION
**Pull request by RattMosss**
Added test cases for a valid header and non-valid header for is_valid_hdr

### CODE REVIEW

**Review by linguini1 — CHANGES_REQUESTED** [review]
Overall looks good! Make sure you resolve the failing workflows and then I'll merge.

File: tests/parsing/test_full_telemetry_parsing.py [hunk 1]
```diff
@@ -1,3 +1,33 @@
...
+
+@pytest.fixture
+def valid_packet_header() -> PacketHeader:
+    # first 32 characters of a packet
+    hdr = "564133494e490000000c010137000000"
+    pkt_hdr = PacketHeader.from_hex(hdr)
+    return pkt_hdr
+@pytest.fixture
+def non_valid_packet() -> PacketHeader:
+    # changed e in valid pakct header fixture to a 2
```
Comment (TASelwyn) [inline tests/parsing/test_full_telemetry_parsing.py:22]: This just changes the call sign from VA3INI to VA3IBI.
I'd recommend adding other tests (such as version 0, length 0, bad block number, all zeroes, etc)

Mainly as just callsign seems like a silly check as you could just add VA3IBI into the approved callsigns list and break this test.

**Review by TASelwyn — CHANGES_REQUESTED** [review]

File: tests/parsing/test_full_telemetry_parsing.py (hunk 1 above)
Comment (EliasJRH) [reply tests/parsing/test_full_telemetry_parsing.py:22]: Seconding this!

Comment (linguini1) [reply tests/parsing/test_full_telemetry_parsing.py:22]: One thing I will say is that the function under test only checks two things:
- Is the call sign recognized?
- Is the version supported?

This function should probably be changed to `from_recognized_callsign` and only report if the call sign is recognized. The version checking should take place when the PacketHeader is constructed and raise an exception if the version isn't supported.

**Review by linguini1 — APPROVED** [review]

**Pull request merged**